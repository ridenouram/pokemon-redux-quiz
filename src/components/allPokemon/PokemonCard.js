import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function PokemonCard({ name, img, color, id }) {
  return (
    <Link to={`/detail/${id}`}>
      <div style={{ background: color, width: '300px', display: 'flex', flexDirection: 'column', alignItems: 'center', margin: '30px' }}>
        <h2>{name}</h2>
        <img src={img} style={{ width: '250px'}} />
      </div>
    </Link>
  );
}

PokemonCard.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default PokemonCard;
