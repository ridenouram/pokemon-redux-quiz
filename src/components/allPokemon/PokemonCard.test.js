import React from 'react';
import { shallow } from 'enzyme';
import PokemonCard from './PokemonCard';

describe('PokemonCard component', () => {
  it('snapshot', () => {
    const wrapper = shallow(<PokemonCard name='bob' color='blue' img='url' />);
    expect(wrapper).toMatchSnapshot();
  });
});
