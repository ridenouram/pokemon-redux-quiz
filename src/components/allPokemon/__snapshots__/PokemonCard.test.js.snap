// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard component snapshot 1`] = `
<Link
  to="/detail/undefined"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "background": "blue",
        "display": "flex",
        "flexDirection": "column",
        "margin": "30px",
        "width": "300px",
      }
    }
  >
    <h2>
      bob
    </h2>
    <img
      src="url"
      style={
        Object {
          "width": "250px",
        }
      }
    />
  </div>
</Link>
`;
